extends Control


@onready var close_button: Button = $Button
@onready var pytanie_label: RichTextLabel = $RichTextLabel
@onready var objasnienie_label: RichTextLabel = $RichTextLabel2

@onready var canvas = %PowitalnyCanvas
@onready var hud = %Hud
			
func _ready() -> void:
	pytanie_label.bbcode_enabled = true
	pytanie_label.text = """
[font_size=12]Gra stworzona na konkurs:[/font_size]
[font_size=14][color=#FFD700]BÅ‚Ä™kitny krwiobieg Ziemi[/color][/font_size]

[font_size=12]Autorzy:[/font_size]
[font_size=14][b]Piotr PryÄ‡[/b][/font_size]
[font_size=14][b]Tomasz MaÅ›lanka[/b][/font_size]

[font_size=14][b]SzkoÅ‚a Podstawowa
numer 83 we WrocÅ‚awiu[/b][/font_size]
	"""

	objasnienie_label.bbcode_enabled = true
	objasnienie_label.text = """
[font_size=14][b][color=#FFD700]ðŸŒŠ Witaj w wodnej przygodzie! ðŸŒŠ[/color][/b][/font_size]

[font_size=12][color=#FFFFFF]Wcielasz siÄ™ w bobra, ktÃ³ry prÃ³buje [b]posprzÄ…taÄ‡ swÃ³j zanieczyszczony dom[/b].

Twoim zadaniem jest zbieraÄ‡ [b]ciekawostki geologiczne i wodne[/b], oraz ratowaÄ‡ Å›rodowisko, [b]usuwajÄ…c Å›mieci z wody i planety![/b][/color][/font_size]
[font_size=10][color=#FFFFFF]<przewiÅ„ w dÃ³Å‚>[/color][/font_size]

[font_size=12][color=#FFFFFF]ðŸ•’ [b]Gra trwa 30 sekund[/b], ale moÅ¼esz przedÅ‚uÅ¼yÄ‡ czas, zbierajÄ…c Å›mieci![/color][/font_size]

[font_size=12][color=#FFFFFF]ðŸŽ® [b]Sterowanie:[/b][/color][/font_size]
[font_size=12][b][color=#FFD700]â† â†’[/color][/b][color=#FFFFFF] - Ruch w lewo i prawo[/color][/font_size]
[font_size=12][b][color=#FFD700]â†‘ â†“[/color][/b][color=#FFFFFF] - Ruch w gÃ³rÄ™ i dÃ³Å‚[/color][/font_size]
[font_size=12][b][color=#FFD700]ESC[/color][/b][color=#FFFFFF] - Pauza[/color][/font_size]

[font_size=12][color=#FFFFFF]ðŸŽ¯ [b]Cel gry:[/b] Zbierz [b]30 ciekawostek[/b] zanim skoÅ„czy siÄ™ czas!  
Gra koÅ„czy siÄ™, gdy:[/color][/font_size]
[font_size=12][color=#FFFFFF]â€¢ UpÅ‚ynie czas  
â€¢ Zbierzesz wszystkie ciekawostki
â€¢ Wejdziesz na stworka[/color][/font_size]

[font_size=12][color=#FFFFFF]Powodzenia! ðŸ¦«ðŸŒ[/color][/font_size]
	"""
	
	canvas.visible = true
	visible = true
	close_button.pressed.connect(_on_close_pressed)
	hud.is_paused = true
	

# Function to show the entire popup window
func show_popup() -> void:
	canvas.visible = true
	visible = true
	hud.is_paused = true

# Function to hide the entire popup window
func _on_close_pressed() -> void:
	canvas.visible = false
	visible = false
	hud.is_paused = false
	

func toggle_visible() -> void:
	canvas.visible = !canvas.visible
	visible = !visible
